### PR TITLE
[WIP] Some Modifications to the Force Field Projector

### PR DESCRIPTION
**Pull request by wesoda25**
## About The Pull Request

Force Field Projector has had its range decreased from 7 (anywhere on screen) to 2. Also, its easier to break down.

## Why It's Good For The Game

Theres been a spike of people using these solely to valid hunt, aka if a traitor does anything they trap them with these and call security. Since the fields have a a very high amount of health, it often takes the traitor too long to escape, and ruins their game unfairly. Lowering the range should limit this behavior, and make sure it only does what is intended (engi/atmos related stuff this item is pretty stupid to begin with). 

## Changelog
:cl:
balance: Force Field Projector now has a range of 2 tiles as opposed to 7
balance: Force Field Projector is much less resistant to melee damage
:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
how did you die to this :joy:

**Comment by wesoda25** [comment]
A better solution might to give them a delay in creating, however I opted not to do this because if that was so, there'd be very minimal difference between this and atmos holofan projector.

**Comment by wesoda25** [comment]
> how did you die to this 😂

I won't lie its unironically a i ded PR, doesn't mean it isn't stupid though.

**Comment by Tlaltecuhtli** [comment]
just lower their hp/armor so people may bash em or give them a do after on building it instead

**Comment by wesoda25** [comment]
> just lower their hp/armor so people may bash em or give them a do after on building it instead

even if it was easier to break through, the same cunts who abuse this shit would just take 5 projectors and spam you with them. Also I'm fairly certain its not based off armor but has a chance to despawn each attack.

**Comment by XDTM** [comment]
It's not chance-based, the generator has a health pool shared by its projected forcefields; damaging one damages them all, and forcefields drain health regularly while active.

**Comment by wesoda25** [comment]
> It's not chance-based, the generator has a health pool shared by its projected forcefields; damaging one damages them all, and forcefields drain health regularly while active.

Hmm then I might give the melee armor a negative value, while keeping the range at 2.

**Comment by fizzle7** [comment]
There's the holofan for stopping gases which came first iirc.

**Comment by imsxz** [comment]
this shits insanely tough to break and its really aids especially when someone runs around with 5 or 6 of them and constantly makes new ones.

**Comment by PKPenguin321** [comment]
Just make it so you can walk through them, like with holo projectors?

**Comment by wesoda25** [comment]
> Just make it so you can walk through them, like with holo projectors?

Then what would be the point of it in the first place?

**Comment by PKPenguin321** [comment]
To keep air in or out, like holoprojectors?

**Comment by wesoda25** [comment]
Holoprojectors? If we let people pass through it’s hardly different than holofans and should be removed.

**Comment by dr-bee** [comment]
It is the only actual weapon that the engineering department gets access to and it isnt even a freaking offensive weapon. This has ided written all over it.

**Comment by PKPenguin321** [comment]
Do you know what holo projectors are or how they work?

**Comment by wesoda25** [comment]
whoops didnt mean to close this.

dr-bee, this isn't a weapon so your argument is stupid. Its intended to guard certain areas and stop things (gas usually) from entering said areas. It is, essentially, a combination of engineering holo-barrier projector and and atmos holo-fan projector . Whereas normal engineering holoprojector is just a cheap holo barrier, this is deployed instantly, anywhere on screen. It takes about 20-40 seconds to destroy one of these, depending on the weapon you have. If you can't imagine why this is an issue, then I don't know what else to say.

PKP, yes, I was referring to ATMOS holofan projector because I tend to see this as an atmos item (given its color scheme).

**Comment by KomradeSpectre** [comment]
Hi, main user of the Forcefield Projectors, Respii Varenos here.

One) These don't stop gasses from passing through them. They're mainly meant for helping things not get sucked out of depressurized areas.

Two) Adding a do-after to it completely negates the use of the above as you will be constantly pulled or pushed and have to initiate the do-after again and again.

Three) Even though they currently do not suffer any negative consequences from ion blasts, I feel they should. I feel an ion should either drastically reduce the charge of the projector, or drop the shield alltogether alongside its charge. This could be extended to have EMP grenades and blasts drop ones in a person's inventory or in the area of the blast.

Four) This heavily reeks of ided, and it has been mainly if not only me using these to valid hunt on the server.

Five) They let energy based weapons through.

Six) They drain faster the more are out concurrently, and the recharge takes a while and makes the shields produced during this recharge rate a lot more brittle.

Seven) A better proposition to nerfing these are making damage directly translate over to the charge, if it doesn't already do that, subtracting the damage from the charge.

Eight) The fields require the projector to be within sight, though currently there is an issue where you can project fields behind opaque walls, and that should be addressed.

**Comment by wesoda25** [comment]
One) I did not know that, but it hardly effects the PR.
Two) ??? Don't quite understand what you mean
Three) Maybe, but making it be effected by the rarest thing in the game (unless EMPs would be included in this) would mean non-traitor antags would get fucked by it as hard as ever (hiveminds, revs, etc etc)
Four) Between you and some randos ive seen abuse this its exposed how unfair it really can be. 
Five) >implying everyone has an energy weapon on them at any given moment
Six) Unless I'm mistaken, this is easily bypassed by using multiple projectors. Even if it isn't so, it still takes far too long for them to drain. 
Seven) already does
Eight) Being anywhere on a persons screen is unfair considering how they are instantly placed. 

Based off this I came across multiple possible solutions. 
One) Still project anywhere on screen, but delay for projection (would be kinda dumb due to space wind which is why you should be using it most of the time)
Two) Slash the health/armor of the shields (would be easily bypassable by using multiple at a time)
Three) Decrease the range, forcing the user to get within distance if they plan to use these for combat (not the intended use!!!)

I settled upon a combination of two and three.

**Comment by wesoda25** [comment]
Oh I suppose adding a lengthy cooldown on the projector after forcefields have been dispelled would be another plausible solution as well.

**Comment by FrostyFridge** [comment]
i ded

**Comment by optimumtact** [comment]
if there is one single more ided comment on this PR I will merge it without any question.

Argue the points, not the incident.

**Comment by FrostyFridge** [comment]
Alternate idea: Leave the range alone, and have the force fields start out incredibly weak when first projected, then have them ramp up in health over the course of say, a minute or so. The health of all force fields from the same projector would be reset down to the minimum when a new field is projected. This would preserve their use as a potent engineering tool or as a fortification, while preventing them from being used proactively as a weapon of any real consequence. Oranges is a salty bitch of a maintainer who doesn't play the game and loves to flex his mod powers while also being unable or unwilling to handle or respond to any real criticism of his sometimes idiotic decisions. He's the /tg/ version of Fox McCloud, the para maintainer who gradually shat all over the codebase with their horrible decisions and people let him get away with it because he does a decent amount of good work that nobody else can be assed to do. It might be necessary to play around with the force field power-up timings in order to make sure that someone trapped by a freshly placed force field can break out with a single swing of a shitty weapon.

**Comment by KomradeSpectre** [comment]
Simple balance fix. 

- Make fields dissipate when hit by ion or emp. If a container, person, or area around or in the blast has projectors, have them also dissipate or reduce the charge of the fields. This will give EMP and Ion a significant role in combating people that use them the way I have made them infamously known for. 

- Make the fields reduce charge based on damage to any of them subtracted from the total charge (which is 100%). This would mean a weapon at 25 brute will drop a field in four hits, and makes the projector suffer a wind up time to get full strength out of the fields.

- Fix the bug where you can forcefield project through walls if it is within the screenspace of the user, as this is currently a bug that happens.

- Possibly make the fields form to full strength after (x) time, but not require a cast up. This would have to be a short time to be used for its intended purpose which is an emergency block. This would allow the user to stay in movement while projecting, but also visually display the field forming to all other parties.

Things that balance it badly:

- Making a charge up cast. This completely negates the intended usage of the fields in the way I stated above, and even more so if you use them as footholds in space (which is a utilitarian usage of them, likely intended).

**Comment by wesoda25** [comment]
Fuck a good solution I can't code off the top of my head.

I'll try and do it tomorrow.

**Comment by optimumtact** [comment]
Don't bother making any pr's you obviously don't want to play nicely so I'm gitbanning you

**Comment by PKPenguin321** [comment]
I just don't think this is a good fix, if you can even call it that. It essentially guts the intended functionality of this device to the point of uselessness so that a single esoteric situation can't happen.

**Comment by wesoda25** [comment]
@PKPenguin321 Seriously whats the intended use at all. To me its a bizarre item which has no real purpose, but relies upon the creativity of the user for multiple situations which might arise.

**Comment by PKPenguin321** [comment]
- I've seen it thrown up at escape after it gets bombed to make emergency space bridges
- I've seen it used to fend meteors off of solars/escape/the shuttle
- I've seen it used to cover up bomb breaches while they get repaired, though this is a surprisingly rare usage

I could have sworn they blocked atmos, but regardless these are what I've seen them used for and they do their job well as a temporary and emergency fixer-upper when large breaches have happened or are about to happened. Always thought they were neat and it's a shame to see a PR like this kill them off so unceremoniously because they happen to be useful in combat sometimes occasionally rarely 1/10000 rounds. If they start getting spammed in combat because they're OP or something then this might be justified, but I've not seen or heard of using them for combat before this PR.

**Comment by datorangebottle** [comment]
As far as I know, the intended uses are:

1. Stop stuff from flying out of the station during a breach so that you can fix shit.
2. Stop people from going into certain places or leaving certain places temporarily.
3. Clowning around

**Comment by PKPenguin321** [comment]
It really would though, since it wouldn't help at all if you have a breach larger than 2 tiles and it would be useless for long space bridges/emergency meteor shields. Claiming that this change doesn't impact the existing functionality is just being intentionally ignorant

**Comment by wesoda25** [comment]
> It really would though, since it wouldn't help at all if you have a breach larger than 2 tiles and it would be useless for long space bridges/emergency meteor shields. Claiming that this change doesn't impact the existing functionality is just being intentionally ignorant

???????????????????????????????????????????????????????????????????????????????????????????????????????
Dude this doesn't change shit, you still get 3 projected barriers, but with 2 different settings. These never COULD make really long space bridges, one projector only makes 3 force fields at a time, and they despawn after a while too. In fact, my suggested change would be BETTER at fulfilling this, as you can make barriers which don't despawn after a certain amount of time.

**Comment by PKPenguin321** [comment]
Hmm. If you insist, but I'll believe it when I see it I guess

**Comment by KomradeSpectre** [comment]
As the person that has mainly used this, and demonstrated strengths and weaknesses readily, I don't think the proposed changes Wesoda has suggested area a good fit for the functionality of the device. Mess with the range, and it's no better than a holobarrier projector and is a redundancy, add modes and it's no better than making it be more powerful than it is now.  Make it have a wind up time, it's again no better than a holobarrier projector.

This seems more like a rushed idea of a balance change than a well thought out one, so I can't in good faith support this, despite wanting a good balance to come to the device that I use a LOT.

**Comment by optimumtact** [comment]
if you're the one ab-using it all the time how can we trust your viewpoint to be any more balanced than OP's

**Comment by KomradeSpectre** [comment]
> if you're the one ab-using it all the time how can we trust your viewpoint to be any more balanced than OP's

I feel like this is bait, but I'll take it.
I've actually openly suggested multiple times ways to balance it on the server after rounds where I've used it to hunt people. I also tell people weak points about the fields while I'm using them occasionally, and some uses that aren't standard, but are helpful to use it (like using it to slow down pursuers).

I'd think my suggestions as listed above are as much proof as anyone would need to trust my viewpoint on balancing it, as I've found as many uses as I can with the fields, which means I have had the most experience I'd say out of any player with it (save maybe the person who coded them).

**Comment by wesoda25** [comment]
tldr their favorite low cost, low mat tool which can be used for virtually anything is going to be changed.

The range is possibly the worst thing about the projector. Really the more I argue over this and learn about the thing, I don't think it even belongs in the game to begin with. 

I agree, if I were to give it a slowdown it would be no different from roundstart engineering holo-projector, hence why I don't wish to do that. 

EMP weakness is an interesting concept and one I think I'll add. However, it should not be the main argument for them to remain unchanged, like it is with borgs. At the end of the day this is a random engineering techwebs item which acts as ranged/instant wall projectors. It adds, from what I can see, nothing inherently essential or useful to the game. That being said I'd hate to deprive the practical users of the item because players are starting to abuse it.

So basically with all that in mind I've come full circle. What I want is: decreased range, pretty low health, emp weakness. If not that, and say the range is what makes the item actually stand out from others (which I disagree with) then 7 tile range as before, however incredibly weak holobarriers, with emp weakness. 

Also increasing required material amount to something a bit more demanding and rare might prove to be of worth, simply to stop people from printing 8 of these.

**Comment by KomradeSpectre** [comment]
> tldr their favorite low cost, low mat tool which can be used for virtually anything is going to be changed.

I don't think any of my posts were read by you if this is what you think I have issue with.

**Comment by wesoda25** [comment]
I constantly address every point you bring up so yes, I think I'm reading your posts.

Anyways I need to toy around with something I'll reopen this real quick.
